openedx/core/djangoapps/discussions/serializers.py

```diff
@@ -265,6 +265,14 @@ def update(self, instance: DiscussionsConfiguration, validated_data: dict) -> Di
             course_overview_id=instance.context_key,
             type='discussion'
         ).update(is_hidden=not instance.enabled)
+        # do the same for the modulestore representation
+        store = modulestore()
+        course = store.get_course(instance.context_key)
+        for tab in course.tabs:
+            if tab.tab_id == 'discussion':
+                tab.is_hidden = not instance.enabled
+                store.update_item(course, self.context['user_id'])
+                break
```

Comment:
While this change will work, I think it isn't optimal. 
Please look at the _update_course_configuration method on the serializer. It is already instantiating the modulestore and getting the course. There you can check if the instance.enabled value doesn't correspond to the tab.is_hidden value and then set the save flag to have the course save only if there is an actual change in value.